pkg/dbmate/db.go

```diff
@@ -441,6 +453,10 @@ func (db *DB) FindMigrations() ([]Migration, error) {
 				migration.Applied = true
 			}
 
+			if db.Strict && !migration.Applied && migration.Version < latestAppliedMigration {
+				return nil, fmt.Errorf("cannot apply migration `%s` after `%s` in --strict mode, migrations have to be in numerical order", migration.Version, latestAppliedMigration)
```

Comment:
similar to my other comment, I think this error message could be a bit more helpful. we can mirror whatever words are selected for the flag description.

Reply:
Changed the error message to the following:

```
migration `%s` is out of order with already applied migrations, the version number has to be higher than the applied migration `%s` in --strict mode
```